### PR TITLE
Run 1813 allow nfs volume mount

### PR DESCRIPTION
**Pull request by natasharomm**

### CODE REVIEW

File: cmd/job/submit/volume_handler.go [hunk 1]
```diff
@@ -137,3 +139,20 @@ func handleVolumes(args *submitArgs) error {
...
+
+func handleNfsVolumes(args *submitArgs) error {
+	var rebuiltNfsVolumeParams []string
+	for _, joinednfsParams := range args.NfsVolumes {
+		nfsParams := strings.Split(joinednfsParams, ",")
+		if len(nfsParams) != 3 {
+			return fmt.Errorf("the --nfs-volume parameter must be given in the form of <server_url>:<path_to_shared_folder>:<container_path>")
+		}
+
+		joinednfsParams = fmt.Sprintf("%s,%s", joinednfsParams, "runai-nfs-volume")
```
Comment (gshaibi) [inline cmd/job/submit/volume_handler.go:151]: Following the last commit, you won't need that anymore

File: cmd/job/submit/volume_handler.go [hunk 2]
```diff
@@ -137,3 +139,20 @@ func handleVolumes(args *submitArgs) error {
 	}
 	return nil
 }
+
+func handleNfsVolumes(args *submitArgs) error {
```
Comment (gshaibi) [inline cmd/job/submit/volume_handler.go:143]: So that function actually becomes validate only right?

**Review by gshaibi — CHANGES_REQUESTED** [review]

**Review by gshaibi — DISMISSED** [review]

**Comment by gshaibi** [comment]
Thanks for teaching me 🥇

File: cmd/job/submit/submit.go [hunk 3]
```diff
@@ -181,9 +182,9 @@ func (submitArgs *submitArgs) addCommonFlags(fbg flags.FlagsByGroups) {
 	flags.AddBoolNullableFlag(flagSet, &submitArgs.LargeShm, "large-shm", "", "Mount a large /dev/shm device.")
 
 	flagSet = fbg.GetOrAddFlagSet(StorageFlagGroup)
-	flagSet.StringArrayVarP(&(submitArgs.Volumes), "volume", "v", []string{}, "Volumes to mount into the container. Syntax: 'HostPath:ContainerPath:AccessMode[optional]'")
+	flagSet.StringArrayVarP(&(submitArgs.Volumes), "volume", "v", []string{}, "Volumes to mount into the container. If volume is not mounted to host, use in conjunction with --nfs-server. Syntax: 'HostPath:ContainerPath:AccessMode[optional]'")
```
Comment (gshaibi) [inline cmd/job/submit/submit.go:185]: maybe, not a must.
```suggestion
	flagSet.StringArrayVarP(&(submitArgs.Volumes), "volume", "v", []string{}, "Volumes to mount into the container. If volume is not mounted on the host, use in conjunction with --nfs-server. Syntax: 'HostPath:ContainerPath:AccessMode[optional]'")
```

File: cmd/job/submit/submit.go [hunk 4]
```diff
@@ -181,9 +182,9 @@ func (submitArgs *submitArgs) addCommonFlags(fbg flags.FlagsByGroups) {
...
 	flagSet.StringArrayVar(&(submitArgs.PersistentVolumes), "pvc", []string{}, "Mount a persistent volume. Syntax: 'StorageClass[optional]:Size:ContainerMountPath:ro[optional]'")
-	flagSet.StringArrayVar(&(submitArgs.NfsVolumes), "nfs-volume", []string{}, "Mount NFS volumes. Syntax: 'Server,PathToSharedFolder,ContainerPath'")
+	flagSet.StringVar(&(submitArgs.NfsServer), "nfs-server", "", "Mount volumes from this server . Use in conjunction with the -v flag.")
```
Comment (gshaibi) [inline cmd/job/submit/submit.go:187]:
```suggestion
	flagSet.StringVar(&(submitArgs.NfsServer), "nfs-server", "", "Mount volumes from this server. Use in conjunction with the -v flag.")
```

**Review by gshaibi — CHANGES_REQUESTED** [review]
Awesome. So we don't support nfs and local host together right?

File: cmd/job/submit/submit.go (hunk 3 above)
Comment (natasharomm) [reply cmd/job/submit/submit.go:185]: Yaron (as a PM representative) sent me the text, I'm not arguing

**Review by gshaibi — APPROVED** [review]

**Review by itayvallach — APPROVED** [review]

**Pull request merged**